src/browser/provider/built-in/base.js

```diff
@@ -0,0 +1,75 @@
+import browserNatives from 'testcafe-browser-natives';
+import OS from 'os-family';
+import WARNING_MESSAGE from '../../../warnings/message';
+
+
+/*eslint-disable no-undef*/
+function getDataFromBrowser () {
+    return { width: window.innerWidth, height: window.innerHeight, title: document.title };
+}
+/*eslint-disable no-undef*/
+
+const initScript = getDataFromBrowser.toString();
+
+
+export default class BaseBrowserProvider {
```

Comment:
Also, do we need base class here? We do base classes only if they are abstract classes. I don't see here any abstract methods